contracts/staking/fuse/FuseStaking.sol

```diff
@@ -187,136 +257,207 @@ contract FuseStaking is
 		}
 	}
 
+	/**
+	 * @dev This function allows guardian to channel the funds of the community pool.
+	 * @param _to An address of the specific user who should receiver the community pool.
+	 */
 	function acquireCommunityPoolBalance(address _to)
 		external
 		onlyRole(GUARDIAN_ROLE)
 	{
 		require(goodDollar.transfer(_to, communityPoolBalance));
 	}
 
-	function _distributeToUBIAndCommunityPool(
-		uint256 _ubiAmount,
-		uint256 _communityPoolAmount
-	) internal returns (uint256 _gdUBIAmount, uint256 _gdCommunityPoolAmount) {
-		if (_ubiAmount == 0 || _communityPoolAmount == 0) {
-			_gdUBIAmount = 0;
-			_gdCommunityPoolAmount = 0;
+	// An inner function which calculates an amount of days of how many the method
```

Comment:
I think its more: 
An inner function that checks if collect UBI Interest was called today: if true it reverts, if false it updates last collection day to today and returns today.